linalg/tmatrix.hpp

```diff
@@ -474,12 +474,12 @@ template <typename scalar_t, typename layout_t, typename data_t>
 inline scalar_t TDet(const layout_t &a, const data_t &A)
 {
    MFEM_STATIC_ASSERT(layout_t::rank == 2, "invalid rank");
-#ifndef __xlC__
+#if __IBMCPP__ > 1399
```

Comment:
Good catch. Unfortunately this doesn't work for other compilers that do not define __IBMCPP__ altogether. I just pushed another change that simply changes __xlC__ to __bg__, which @v-dobrev and I checked is defined on xlC on Vulcan.